### PR TITLE
Optick + Single Axis Prune

### PR DESCRIPTION
**Pull request by McCallisterRomer**
Can't type, wrist broken...

### CODE REVIEW

File: nc/include/ecs/Entity.h [hunk 1]
```diff
@@ -42,10 +44,12 @@ namespace nc
             constexpr auto Layer() const noexcept { return m_layer; }
             constexpr auto Flags() const noexcept { return m_flags; }
             constexpr auto IsStatic() const noexcept { return m_flags & Flags::Static; }
+            constexpr auto IsPersistent() const noexcept { return m_flags & Flags::Persistent; }
```
Comment (jaremieromer) [inline nc/include/ecs/Entity.h:47]: For Index, Layer, and Flags - the "GetX" conversation

File: nc/source/graphics/techniques/WireframeTechnique.cpp [hunk 2]
```diff
@@ -126,8 +124,6 @@ namespace nc::graphics
 
     void WireframeTechnique::Record(vk::CommandBuffer* cmd, const PerFrameRenderState& frameData)
     {
```
Comment (jaremieromer) [inline nc/source/graphics/techniques/WireframeTechnique.cpp:125]: Don't actually have optick used here in this class

File: nc/source/physics/collision/IntersectionQueries.h [hunk 3]
```diff
@@ -31,6 +20,9 @@ namespace nc::physics
     bool Intersect(const Sphere& a, const Sphere& b);
     bool Intersect(const Frustum& a, const Sphere& b);
 
+    // check, also must be aabb
```
Comment (McCallisterRomer) [inline nc/source/physics/collision/IntersectionQueries.h]: What am I trying to tell myself?

File: project/source/collision_events/CollisionEvents.cpp [hunk 4]
```diff
@@ -46,6 +46,8 @@ namespace nc::sample
 {
     void CollisionEvents::Load(NcEngine* engine)
     {
+        prefab::InitializeResources();
```
Comment (McCallisterRomer) [inline project/source/collision_events/CollisionEvents.cpp]: can remove

File: nc/source/physics/dynamics/Dynamics.cpp [hunk 5]
```diff
@@ -2,7 +2,7 @@
 #include "ecs/component/Collider.h"
 #include "ecs/component/PhysicsBody.h"
 #include "physics/PhysicsConstants.h"
-#include "debug/Profiler.h"
+#include "optick/optick.h"
```
Comment (McCallisterRomer) [inline nc/source/physics/dynamics/Dynamics.cpp]: can remove

File: nc/source/physics/dynamics/Solver.cpp [hunk 6]
```diff
@@ -1,12 +1,13 @@
 #include "Solver.h"
+#include "ecs/Registry.h"
 #include "physics/PhysicsConstants.h"
 #include "graphics/DebugRenderer.h"
-#include "debug/Profiler.h"
-//#define USE_DEBUG_RENDERING
-#undef USE_DEBUG_RENDERING
+#include "optick/optick.h"
```
Comment (McCallisterRomer) [inline nc/source/physics/dynamics/Solver.cpp]: can remove

File: nc/source/physics/collision/broad_phase/SingleAxisPrune.h [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        m_sortedEntries.shrink_to_fit();
+    }
+
+    template<ProxyCache ProxyCacheType>
+    template<SingleAxisPrune<ProxyCacheType>::Axis SortAxis>
+    float SingleAxisPrune<ProxyCacheType>::GetMin(const Sphere& estimate)
+    {
+        if constexpr(SortAxis == SingleAxisPrune<ProxyCacheType>::Axis::X)
+            return estimate.center.x - estimate.radius;
+        else if(SortAxis == SingleAxisPrune<ProxyCacheType>::Axis::Y)
```
Comment (jaremieromer) [inline nc/source/physics/collision/broad_phase/SingleAxisPrune.h]: can this be constexpr as well?

File: nc/source/physics/collision/broad_phase/SingleAxisPrune.h [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+            return estimate.center.z - estimate.radius;
+    }
+
+    template<ProxyCache ProxyCacheType>
+    template<SingleAxisPrune<ProxyCacheType>::Axis SortAxis>
+    float SingleAxisPrune<ProxyCacheType>::GetMax(const Sphere& estimate)
+    {
+        if constexpr(SortAxis == SingleAxisPrune<ProxyCacheType>::Axis::X)
+            return estimate.center.x + estimate.radius;
+        else if(SortAxis == SingleAxisPrune<ProxyCacheType>::Axis::Y)
```
Comment (jaremieromer) [inline nc/source/physics/collision/broad_phase/SingleAxisPrune.h]: can this be constexpr as well?

File: project/source/spawn_test/SpawnTest.cpp [hunk 9]
```diff
@@ -4,42 +4,57 @@
...
+        
+        ImGui::EndChild();
     }
 }
 
 namespace nc::sample
 {
     void SpawnTest::Load(NcEngine* engine)
     {
+        prefab::InitializeResources();
```
Comment (McCallisterRomer) [inline project/source/spawn_test/SpawnTest.cpp]: can remove

File: nc/source/physics/PhysicsPipelineTypes.cpp [hunk 10]
```diff
@@ -1,10 +1,13 @@
-#include "Manifold.h"
+#include "PhysicsPipelineTypes.h"
+#include "debug/Utils.h"
 #include "ecs/Registry.h"
+#include "ecs/component/PhysicsBody.h"
 #include "physics/PhysicsConstants.h"
-#include "debug/Utils.h"
 
 #include <array>
```
Comment (jaremieromer) [inline nc/source/physics/PhysicsPipelineTypes.cpp:8]: Extra newline

File: nc/source/physics/PhysicsPipelineTypes.cpp [hunk 11]
```diff
@@ -1,10 +1,13 @@
...
 
+#include <iostream>
```
Comment (jaremieromer) [inline nc/source/physics/PhysicsPipelineTypes.cpp]: Possibly don't need - std::cerr is commented out

**Review by jaremieromer — APPROVED** [review]

**Pull request merged**